src/utils/c_utils/src/string_c_utils.c

```diff
@@ -92,12 +92,12 @@ bool ReadDelimitedValueFromFile(const char* fileName, const char* key, char* val
  * @brief Function that sets @p strBuffers to the contents of the file at @p filePath if the contents are smaller in size than the buffer
  * @param filePath path to the file who's contents will be read
  * @param strBuffer buffer which will be loaded with the contents of @p filePath
- * @param strBuffSize the size of the buffer
+ * @param strBuffSize the size of the buffer to hold file contents plus a null terminator.
  * @returns false on failure, true on success
  */
 bool LoadBufferWithFileContents(const char* filePath, char* strBuffer, const size_t strBuffSize)
```

Comment:
looking at usage patterns, we seem to always do:

```c
        if (!LoadBufferWithFileContents(config.edgegatewayCertPath, certificateString, ARRAY_SIZE(certificateString)))
        {
            Log_Error("Failed to read the certificate from path: %s", config.edgegatewayCertPath);
            goto done;
        }

        if (mallocAndStrcpy_s(&info->certificateString, certificateString) != 0)
```

so there's no reason this method shouldn't do an allocation -- it's much safer to do (pseudo-code, add necessary error checking):

```c
// Caller must call free on returned pointer.
char* LoadBufferWithFileContents(const char* filePath)
{
  size = stat(filePath)
  buf = alloc(size);
  read(filePath, buf, size)
  return buf
}
```

Reply:
and in the case above, just do:

```c
info->certificateString = LoadBufferWithFileContents(config.edgegatewayCertPath);
if (info->certificateString == NULL) 
{
    Log_Error...
```

Reply:
:thinking::  Could also use `mmap` to avoid the memory allocation? **Assuming the file content is static.**